internals/daemon/daemon.go

```diff
@@ -58,6 +58,44 @@ var (
 	systemdSdNotify = systemd.SdNotify
 )
 
+// ApiRequestSrc defines the possible API request sources that we support.
+// This can be extracted from http.Request using RequestSrc.
+type ApiRequestSrc int
+
+const (
+	ApiRequestSrcCtxKey                = ApiRequestSrcUnknown
+	ApiRequestSrcUnknown ApiRequestSrc = iota
+	ApiRequestSrcUnixSocket
+	ApiRequestSrcHTTP
+	ApiRequestSrcHTTPS
+)
+
+// RequestSrc extracts the source of the HTTP request. If the source
+// cannot be found in the context, it returns ApiRequestSrcUnknown.
+func RequestSrc(r *http.Request) ApiRequestSrc {
```

Comment:
```suggestion
func RequestSource(r *http.Request) ApiRequestSrc {
```

Reply:
I opted for ```TransportType```, and did a mayor refactor. Let me know what you think ?

Reply:
Resolving, but feel free to reopen.